include/arch/arm/aarch32/irq.h

```diff
@@ -107,12 +105,16 @@ extern void z_arm_interrupt_init(void);
  */
 #define ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p) \
 { \
+	BUILD_ASSERT(!(!IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS) && (flags_p & IRQ_ZERO_LATENCY)), \
```

Comment:
This double negation looks a bit odd. Maybe you could rephrase this logic (use `a || !b` instead of `!(!a && b)`) to make it clearer?

Reply:
done